redash/models.py

```diff
@@ -945,6 +1150,13 @@ def recent(cls, org, groups, user_id, for_user=False, limit=20):
     def get_by_slug_and_org(cls, slug, org):
         return cls.get(cls.slug == slug, cls.org==org)
 
+    def tracked_save(self, changing_user, old_object=None, *args, **kwargs):
+        self.version += 1
+        self.save(*args, **kwargs)
+        # save Change record
+        new_change = Change.save_change(user=changing_user, old_object=old_object, new_object=self)
+        return new_change
```

Comment:
Instead of duplicating the `tracked_save` logic, let's have a mixin for this like we have for timestamps. 

Also I'm not happy with the added complexity of the way the actual change is tracked. We currently have no use for the change itself, how about we don't implement this as part of this pull request?